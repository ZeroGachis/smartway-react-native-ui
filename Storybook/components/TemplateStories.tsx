import type { Meta, StoryObj } from '@storybook/react-native';
import React from 'react';
import { StyleSheet, View } from 'react-native';

// fake component must be replace by one from the library
const Component = (props: React.PropsWithChildren) => {
    const { children } = props;
    return <>{children}</>;
}; // must be removed

type ComponentProps = React.ComponentProps<typeof Component>;

export default {
    title: 'components/Component', // change it with your component name eg('components/AwesomeButton',)
    component: Component,
    //default value
    args: {},
    // add some control in argTypes
    // some example
    // https://github.com/storybookjs/react-native/blob/next/examples/expo-example/components/ControlExamples/ControlExample/ControlExample.stories.tsx
    argTypes: {},
    // use it to wrap your storie in composant
    decorators: [
        (Story) => {
            const styles = StyleSheet.create({
                container: { paddingTop: 16 },
            });
            return (
                <View style={styles.container}>
                    <Story />
                </View>
            );
        },
    ],
} as Meta<ComponentProps>;

type Story = StoryObj<ComponentProps>;

// first story use it with args to change load only props value
// you can also use render as
// render : (args) => <AwesomeButton {...args}>
export const Default: Story = {
    args: {},
};
Default.parameters = { noSafeArea: false };
