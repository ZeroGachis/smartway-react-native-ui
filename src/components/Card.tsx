import type { ReactNode } from "react";
import React from "react";
import { StyleProp, View, ViewStyle } from "react-native";
import { useTheme } from "smartway-react-native-ui";

interface Props {
    children?: ReactNode,
    style: ViewStyle
};

export const Card = ({children, style}: Props) => {
    const theme = useTheme();
    const cardStyle: StyleProp<ViewStyle> = {
        backgroundColor: theme.sw.colors.neutral[50],
        width: '100%',
        alignItems: 'center',
        borderRadius: 12,
        paddingTop: theme.sw.spacing.xl,
        paddingBottom: theme.sw.spacing.xl,
        paddingLeft: theme.sw.spacing.m,
        paddingRight: theme.sw.spacing.m,
        ...style
        // TODO: Implement shadows
    };

    return (
        <View style={cardStyle}>
            {children}
        </View>
    );
};
