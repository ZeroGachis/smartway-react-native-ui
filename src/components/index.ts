import { Button } from 'react-native-paper';
import Logo from './logos/logos';

export { Button, Logo };