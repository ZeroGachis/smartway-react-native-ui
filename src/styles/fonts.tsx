export const fontConfig = {
    fontFamily: 'Roboto'
};