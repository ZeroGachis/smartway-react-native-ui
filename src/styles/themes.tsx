import React, { ReactNode } from 'react';
import {
    configureFonts,
    MD3LightTheme,
    Provider as PaperProvider,
    useTheme as usePaperTheme,
} from 'react-native-paper';
import { fontConfig } from './fonts';

const theme = {
    ...MD3LightTheme,
    fonts: configureFonts({ config: fontConfig }),
    sw: {
        colors: {
            primary: {
                800: '#225348',
                600: '#017F63',
                400: '#18A586',
                200: '#31E3BC',
                50: '#C5FCF0',
            },
            neutral: {
                900: '#1A2026',
                800: '#212B36',
                700: '#2F4158',
                600: '#637381',
                500: '#919EAB',
                400: '#E5EAEE',
                300: '#ECF0F4',
                200: '#F4F6F8',
                100: '#F9FAFB',
                50: '#FFFFFF',
            },
            error: {
                main: '#FF5630',
                50: '#FFE9D5',
            },
        },
        spacing: {
            xs: 6,
            s: 12,
            m: 16,
            l: 24,
            xl: 32,
            xxl: 64,
        },
    },
};

export type ThemType = typeof theme;
export const useTheme = () => usePaperTheme<ThemType>();

type Props = {
    children?: ReactNode;
};

export const ThemeProvider = ({ children }: Props) => {
    return <PaperProvider theme={theme}>{children}</PaperProvider>;
};
