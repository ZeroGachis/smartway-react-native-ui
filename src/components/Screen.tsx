import React from 'react';
import { SafeAreaView, ViewStyle, StatusBar, StyleSheet } from 'react-native';
import { useSafeAreaInsets } from 'react-native-safe-area-context';
import { useTheme } from '../styles/themes';
import DeviceInfo from 'react-native-device-info';
import type { WithTestID } from 'src/shared/type';

type Props = WithTestID<{
    children?: React.ReactNode;
    style?: ViewStyle;
    statusBarColor?: string;
}>;

export const Screen = ({ children, style, testID, statusBarColor }: Props) => {
    const theme = useTheme();
    const insets = useSafeAreaInsets();
    const isTablet = DeviceInfo.isTablet();

    const styles = StyleSheet.create({
        screen: {
            flex: 1,
            backgroundColor: theme.sw.color.neutral[0],
            marginTop: insets.top,
            paddingLeft: isTablet ? theme.sw.spacing.s : 0,
            paddingRight: isTablet ? theme.sw.spacing.s : 0,
            ...style,
        },
    });

    return (
        <SafeAreaView style={styles.screen} testID={testID}>
            <StatusBar
                translucent
                backgroundColor={
                    statusBarColor
                        ? statusBarColor
                        : theme.sw.color.neutral[0]
                }
                barStyle='dark-content'
            />
            {children}
        </SafeAreaView>
    );
};
