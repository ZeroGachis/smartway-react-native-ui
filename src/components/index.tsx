import { Button } from './buttons/Button';
import { IconButton } from './buttons/IconButton';
import { Logo } from './logos/Logo';
import { Screen } from './Screen';
import { Headline } from './typography/Headline';
import { Body } from './typography/Body';
import { Modal } from './modals/Modal';
import { TextField } from './fields/TextField';
import { Icon } from './icons/Icon';
import { TopAppBar } from './topAppBar/TopAppBar';
import { Card } from './card/Card';
import { Tag } from './tag/Tag';
import { Divider } from './divider/Divider';
import { LineChart } from './charts/LineChart';
import { Badge } from './badge/Badge';
import { DateField } from './dateSelector/DateField';
import { DateSelector } from './dateSelector/DateSelector';
import { Label } from './typography/Label';
import { FieldState, NumberField } from './numberField/NumberField';

export { default as Snackbar, useSnackbar } from './alert/Snackbar';
export { default as Banner, useBanner } from './alert/Banner';

export {
    Body,
    Button,
    IconButton,
    Modal,
    Headline,
    Icon,
    Logo,
    Label,
    Screen,
    TextField,
    TopAppBar,
    NumberField,
    FieldState,
    Card,
    Tag,
    Divider,
    LineChart,
    Badge,
    DateField,
    DateSelector,
};
