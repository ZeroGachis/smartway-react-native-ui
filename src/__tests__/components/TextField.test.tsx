import React from 'react';
import { fireEvent, render, act } from '@testing-library/react-native';
import { ThemeProvider } from '../../styles/themes';
import { TextField } from '../../components';

const mockedCallback = jest.fn();
const mockedTestID = 'mockedTestID';

describe('MODULE | TextField', () => {
    it('component renders correctly', () => {
        const tree = render(
            <ThemeProvider>
                <TextField textType="information" value="" />
            </ThemeProvider>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('component renders correctly with bottom text', () => {
        const tree = render(
            <ThemeProvider>
                <TextField textType="information" value="" text="test" />
            </ThemeProvider>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('component renders correctly when textType in error state', () => {
        const tree = render(
            <ThemeProvider>
                <TextField textType="error" value="" />
            </ThemeProvider>,
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('fires callback when on input value change', async () => {
        const tree = render(
            <ThemeProvider>
                <TextField
                    testID={mockedTestID}
                    textType="information"
                    value=""
                    onChangeText={mockedCallback}
                />
            </ThemeProvider>,
        );

        await act(async () => {
            fireEvent.changeText(tree.getByTestId(mockedTestID), '12345');
        });

        expect(mockedCallback).toHaveBeenCalledWith('12345');
    });
});
