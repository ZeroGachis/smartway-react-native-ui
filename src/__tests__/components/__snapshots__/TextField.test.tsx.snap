// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MODULE | TextField component renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        pointerEvents="none"
        style={
          {
            "marginBottom": 24,
          }
        }
      >
        <View
          style={
            {
              "fontFamily": "PublicSans-Regular",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#2F4158",
                  "borderRadius": 8,
                  "borderWidth": 1,
                },
                {
                  "borderColor": "#C4CDD5",
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                [
                  {
                    "paddingBottom": 0,
                  },
                  {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                cursorColor="#212B36"
                editable={true}
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholderTextColor="#919EAB"
                selectionColor="#212B36"
                style={
                  [
                    {
                      "margin": 0,
                      "zIndex": 1,
                    },
                    {
                      "height": 56,
                    },
                    {
                      "paddingBottom": 18.5,
                      "paddingTop": 18.5,
                    },
                    {
                      "color": "#1A2026",
                      "fontFamily": "PublicSans-Regular",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": 19,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    [
                      {},
                    ],
                    undefined,
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`MODULE | TextField component renders correctly when textType in error state 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        pointerEvents="none"
        style={
          {
            "marginBottom": 24,
          }
        }
      >
        <View
          style={
            {
              "fontFamily": "PublicSans-Regular",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#FF5630",
                  "borderRadius": 8,
                  "borderWidth": 2,
                },
                {
                  "borderColor": "#FF5630",
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                [
                  {
                    "paddingBottom": 0,
                  },
                  {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                cursorColor="#FF5630"
                editable={true}
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholderTextColor="#919EAB"
                selectionColor="#FF5630"
                style={
                  [
                    {
                      "margin": 0,
                      "zIndex": 1,
                    },
                    {
                      "height": 56,
                    },
                    {
                      "paddingBottom": 18.5,
                      "paddingTop": 18.5,
                    },
                    {
                      "color": "#1A2026",
                      "fontFamily": "PublicSans-Regular",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": 19,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    [
                      {},
                    ],
                    undefined,
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`MODULE | TextField component renders correctly with bottom text 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        pointerEvents="none"
        style={
          {
            "marginBottom": 24,
          }
        }
      >
        <View
          style={
            {
              "fontFamily": "PublicSans-Regular",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              [
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#2F4158",
                  "borderRadius": 8,
                  "borderWidth": 1,
                },
                {
                  "borderColor": "#C4CDD5",
                  "borderWidth": 1,
                },
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                [
                  {
                    "paddingBottom": 0,
                  },
                  {
                    "minHeight": 56,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <TextInput
                cursorColor="#212B36"
                editable={true}
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholderTextColor="#919EAB"
                selectionColor="#212B36"
                style={
                  [
                    {
                      "margin": 0,
                      "zIndex": 1,
                    },
                    {
                      "height": 56,
                    },
                    {
                      "paddingBottom": 18.5,
                      "paddingTop": 18.5,
                    },
                    {
                      "color": "#1A2026",
                      "fontFamily": "PublicSans-Regular",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "letterSpacing": 0.15,
                      "lineHeight": 19,
                      "paddingHorizontal": 16,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    [
                      {},
                    ],
                    undefined,
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginTop": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#637381",
                "marginLeft": 4,
              }
            }
          >
            test
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
